File: frontend/src/types/booking.ts
export type Booking = {
	id: string;
	date: Date;
	description: string;
	amount: number;
	category: string;
	type: Type;
	fileUrl?: string;
}

export type Type = 'INCOME' | 'EXPENSE';

export type BookingForm = {
	date: Date;
	description: string;
	amount: number;
	category: string;
	type: Type;
	fileUrl?: string;
}

File: backend/src/main/java/de/webdev/backend/cashjournal/models/Booking.java
package de.webdev.backend.cashjournal.models;

import lombok.With;
import org.springframework.data.mongodb.core.mapping.MongoId;

import java.time.LocalDate;

@With
public record Booking(
		@MongoId
		String id,
		LocalDate date,
		String description,
		double amount,
		String category,
		Type type,
		String fileUrl) {

}